scripts/buf.pl

```diff
@@ -40,64 +43,59 @@
 sub _filename { sprintf '%s/scrollbuffer', get_irssi_dir }
 
 sub upgrade {
-    my $fn = _filename;
-    my $old_umask = umask 0077;
-    open my $fh, q{>}, $fn or die "open $fn: $!";
-    umask $old_umask;
-
-    print $fh join("\0", map $_->{server}->{address} . $_->{name}, channels), "\n";
+    my $out = { suppress => join("\0", map $_->{server}->{address} . $_->{name}, channels) };
```

Comment:
how about you replace this with a structure / array instead of the awkward NUL byte